### ISSUE TITLE
Add more classifier

### ISSUE THREAD

**Issue by poig** [qiskit-community/qiskit-machine-learning#492]
### What should we add?

we should include more classifiers:

- [ ] Quantum K-NN has already been done, but only fit the old version. 
https://github.com/GroenteLepel/qiskit-quantum-knn
- [ ] Quantum Random Forest Classifier
https://arxiv.org/pdf/2210.02355.pdf
https://github.com/maiyuren/Quantum-Random-Forest

Maybe we should add these also:
- [ ] Quantum GaussianNB
- [ ] Quantum DecisionTreeClassifier

**Comment by chin-jey** [comment]
There's this project implementing other quantum classification and regression methods : https://github.com/mspronesti/qlearnkit

**Comment by poig** [comment]
> 

Yeah, include K-NN classification, regression, dataset, encoding map, swaptest circuit, but it's not fit the latest version.

**Comment by chin-jey** [comment]
Why not ? It seems it does to me.

**Comment by poig** [comment]
> Why not ? It seems it does to me.
It still uses the base backend

**Comment by poig** [comment]
> Why not ? It seems it does to me.

I just see that it fit with the sources code, not the latest release version.

**Comment by adekusar-drl** [comment]
For example, quantum KNN requires either access to quantum memory to keep the states handy or re-creation of states every time when a distance is evaluated. Thus, the quantum version of the algorithm does not bring any advantages over the classical version. I can't say the above mentioned algorithms suffer from the same problem, they have to be studied and evaluated if they can bring value for this package. If you want to develop one of them, and possibly to contribute them, we can discuss in more details.

**Comment by edoaltamura** [comment]
Hi @poig @chin-jey, have there been recent developments on this discussion? Feel free to post a simple example here, as it might be useful for others. Once we've seen an implementation, we can evaluate together if it's worth turning it into a PR.

**Comment by JulioBuzzi** [comment]
Hey @edoaltamura, is this issue still open and available for possible future PRs?

**Comment by oscar-wallis** [comment]
@JulioBuzzi  Any updates on this issue?
